### PR TITLE
[nrf52xx] notify if ACK was sent with frame pending bit set

### LINKED ISSUE #3826: [nrf52xx] set acknowledged with frame pending flag
**Is your feature request related to a problem? Please describe.**
Needs set acked frame pending flag in radio driver

**Describe the solution you'd like**
Implement in radio Nordic SDK.

**Describe alternatives you've considered**
None

**Additional context**
See more in #3785

### PR DESCRIPTION
**Pull request by LuDuda**
This PR should fix #3826

### CODE REVIEW

File: examples/platforms/nrf52811/radio.c [hunk 1]
```diff
@@ -87,6 +90,7 @@ static otRadioIeInfo sReceivedIeInfos[NRF_802154_RX_BUFFERS];
 static otInstance *sInstance = NULL;
 
 static otRadioFrame sAckFrame;
+static bool         sAckedWithFramePending = true;
```
Comment (bukepo) [inline examples/platforms/nrf52811/radio.c]: Should the flag be initialized as false, which indicates two situations?

The frame pending in the sent ACK frame is not set
No ACK was sent for the received frame, either for not requested or error.
Thought?

**Comment by KaSroka** [comment]
LGTM :+1:

File: examples/platforms/nrf52811/radio.c [hunk 2]
```diff
@@ -738,6 +745,8 @@ void nrf_802154_received_raw(uint8_t *p_data, int8_t power, uint8_t lqi)
     receivedFrame->mIeInfo->mTimestamp = otPlatTimeGet() - offset;
 #endif
 
+    sAckedWithFramePending = true;
```
Comment (bukepo) [inline examples/platforms/nrf52811/radio.c]: Shall we set as false?

File: examples/platforms/nrf52811/radio.c [hunk 3]
```diff
@@ -770,9 +779,17 @@ void nrf_802154_receive_failed(nrf_802154_rx_error_t error)
         assert(false);
     }
 
+    sAckedWithFramePending = true;
```
Comment (bukepo) [inline examples/platforms/nrf52811/radio.c]: same as above, set as false?

File: examples/platforms/nrf52811/radio.c (hunk 1 above)
Comment (LuDuda) [reply examples/platforms/nrf52811/radio.c]: I'm fine with this approach, my assumption was that if ACK was not sent e.g. due to BLE activity in multiprotocol configuration, we prefer to set this flag to 1, but I think you are right, that setting it to 0 is better.

Comment (LuDuda) [reply examples/platforms/nrf52811/radio.c]: I also decided to keep redundant if-statement and check if the frame has ACK-requested bit set for readability in `nrf_802154_received_raw` function.

**Pull request merged**